solvebio/global_search.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from .client import client
+from .errors import SolveError
+from .query import QueryBase
+from .query import Query
+from .query import Filter
+
+import logging
+logger = logging.getLogger('solvebio')
+
+
+class GlobalSearch(Query):
+    """
+    GlobalSearch acts as a request wrapper that generates a request from Filter objects,
+    and can iterate through streaming result sets.
+    """
+
+    def __init__(
+            self,
+            query=None,
+            filters=None,
+            entities=None,
+            ordering=None,
+            limit=float('inf'),
+            page_size=QueryBase.DEFAULT_PAGE_SIZE,
+            result_class=dict,
+            debug=False,
+            **kwargs):
+        """
+        Creates a new Query object.
+
+        :Parameters:
+          - `query` (optional): An optional query string (advanced search).
+          - `filters` (optional): Filter or List of filter objects.
+          - `entities` (optional): List of entity tuples to filter on (entity type, entity).
+          - `ordering` (optional): List of fields to order the results by.
+          - `limit` (optional): Maximum number of query results to return.
+          - `page_size` (optional): Number of results to fetch per query page.
+          - `result_class` (optional): Class of object returned by query.
+          - `debug` (optional): Sends debug information to the API.
+        """
+        self._data_url = '/v2/search'
```

Comment:
Call super __init__ and then overwrite the data_url that way

Reply:
I know that `GlobalSearch` and `Query` classes are not quite the same, but they share a lot of common functionalities. I still believe that GlobalSearch should be derived from Query class, so we don't unnecessarily duplicate the code that's already in there.

Reply:
Yes I agree, just trying to further remove some duplicate code. That approach above sounds fine to me